demo_modules/big_image_carousel/big_image_carousel.js

```diff
@@ -0,0 +1,199 @@
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+const ModuleInterface = require('lib/module_interface');
+const debug = require('debug');
+const register = require('register');
+const wallGeometry = require('wallGeometry');
+
+class BigImageCarouselServer extends ModuleInterface.Server {
+  constructor(config) {
+    super();
+    debug('BigImageCarousel Server!', config);
+  }
+}
+
+// Create a polygon extending from x,y by w,h.
+// p5 must be a P5.js instance.
+class BigImageCarouselSketch {
+  constructor(p5, surface, extraArgs) {
+    this.p5 = p5;
+    this.surface = surface;
+    this.tiles = null; 
+
+    this.image_list_config = extraArgs;
+  }
+
+  preload() {
+    var p5 = this.p5;
+
+    console.log('preload');
+
+    this.y_index = this.surface.virtualRect.y / this.surface.virtualRect.h;
+
+    console.log('screen with y_index ' + this.y_index);
+
+    const asset = require('client/asset/asset');
+
+    this.tiles = new Array(); 
+    for (let path in this.image_list_config) {
+      console.log('Path is ' + path);
+      let images = this.image_list_config[path];
+      for (let i in images) {
+        let image = images[i];
+        for (let x = 0; x < image.num_x_tiles; x++) {
+          let tilePath = asset(path + image.name + '_tile_' + x + '_' + this.y_index + '.' + image.extension);
+          console.log('tilePath is ' + tilePath);
+          this.tiles.push(p5.loadImage(tilePath));
+        }
+      }
+    }
+  }
+
+  setup() {
+    var p5 = this.p5;
+
+    console.log('setup()');
+
+    p5.textSize(40);
+    p5.fill(200, 0, 100);
+    // Each instance will only ever draw images from one Y offset, but many X offsets.
+
+    this.sum_tile_width = 0;
+    for (let x = 0; x < this.tiles.length; x++) {
+      this.sum_tile_width += this.tiles[x].width;
+    }
+    console.log('num tiles is ' + this.tiles.length);
+    console.log('sum_tile_width is ' + this.sum_tile_width);
+    if (this.sum_tile_width < this.surface.wallRect.w) {
+      let initial_sum_tile_width = this.sum_tile_width;
+      while (this.sum_tile_width < this.surface.wallRect.w) {
+        this.tiles = this.tiles.concat(this.tiles);
```

Comment:
it basically extends the tile list to have enough tiles to be wider than the screen.